### PR TITLE
Stop robot motion when it moves out of 2nd geofence 

### PR DESCRIPTION
**Pull request by iche033**
Signed-off-by: Ian Chen <ichen@osrfoundation.org>

There are 2 geofences for the competition area. If a robot moves outside of the first geofence, a time penalty is given (implemented in https://github.com/osrf/mbzirc/pull/41). 

This PRs adds the 2nd / outer geofence. If a robot moves outside of this geofence, ~a `HaltMotion` component is added to the robot model which will lock its joints and stop its motion - similar to what a 'kill switch' does in the real world.~  the robot is made static. 
* Note: I ran into a crash with ode collision checking error when quadrotor fell from sky and crashing into boundary collisions at high speed, so now just freezing the robot instead of locking its joints with `HaltMotion` component

To test:

1. launch coast world and spawn a quadrotor
    ```sh
    # launch coast world
    ros2 launch ros_ign_gazebo ign_gazebo.launch.py ign_args:="-v 4 -r coast.sdf"
    # spawn quadrotor
    ros2 launch mbzirc_ign spawn.launch.py name:=quadrotor_1 world:=coast model:=mbzirc_quadrotor type:=uav x:=-1500 y:=0 z:=3.6 R:=0 P:=0 Y:=0
    ```

1. Follow the quadrotor by right clicking on the quadrotor model in the 3D window and select `Follow`

1. In a separate terminal, monitor `events.yml` log

    ```
    tail -f /tmp/ign/mbzirc/logs/events.yml
    ```

1. Make the quadrotor take off

    ```
    ros2 topic pub --once /quadrotor_1/cmd_vel geometry_msgs/msg/Twist '{linear: {x: 0,y: 0.0, z: 2.0}, angular: {x: 0.0, y: 0.0, z: 0.0}}'
    ```

1. Move the robot to z = 100 so that it's close to the ceiling of the first geofence

    ```
    ign service -s /world/coast/set_pose --reqtype ignition.msgs.Pose --reptype ignition.msgs.Boolean --timeout 300 --req 'name: "quadrotor_1", position: {x:-1500, y:0, z: 100.0}'
    ```

1. The quadrotor should still be flying upwards and once it's reaches above ~122m, you should see a `exceed_boundary_1` type event logged in `events.yml` file

1. Let the quadrotor continue to fly upwards and once it reaches ~147m where the ceiling of the 2nd geofence is, you should see a `exceed_boundary_outer` type event logged in `events.yml` file. ~The quadrotor's propellers will also stop rotating and it should fall from the sky~ The quadrotor should should freeze in mid air.

### CODE REVIEW

File: mbzirc_ign/src/GameLogicPlugin.cc [hunk 1]
```diff
@@ -252,6 +256,19 @@ class mbzirc::GameLogicPluginPrivate
...
+  /// is terminated on the third occurence. If a robot moves outside of the
+  /// outer geofence, the robot is disabled
+  /// \param[in] _ecm Mutable reference to Entity Component Manager
+  public: void CheckRobotsInGeofenceBoundary(
+      EntityComponentManager &_ecm);
+
+  /// \brief Make an entity static
+  /// \param[in] _entity Entity to make static
+  /// \param[in] _ecm Mutable reference to Entity Component Manager
+  public: bool MakeStatic(Entity _entity, EntityComponentManager &_ecm);
```
Comment (caguero) [inline mbzirc_ign/src/GameLogicPlugin.cc:270]: Missing `\return`.

File: mbzirc_ign/src/GameLogicPlugin.cc [hunk 2]
```diff
@@ -795,58 +844,8 @@ void GameLogicPlugin::PostUpdate(
...
-          // terminate run
-          this->dataPtr->Finish(this->dataPtr->simTime);
-        }
-      }
-      this->dataPtr->robotsInBoundary[robotEnt] = isInBounds;
-    }
-  }
-
   // find the sensor associated with the input stream
+  // used for valdiating target reports
```
Comment (caguero) [inline mbzirc_ign/src/GameLogicPlugin.cc:848]: typo: validating

File: mbzirc_ign/src/GameLogicPlugin.cc [hunk 3]
```diff
@@ -931,6 +929,134 @@ void GameLogicPlugin::PostUpdate(
...
+      this->robotsInBoundary[robotEnt] = isInBounds;
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+bool GameLogicPluginPrivate::MakeStatic(Entity _entity,
+    EntityComponentManager &_ecm)
+{
+  // make breadcrumb static by spawning a static model and attaching the
```
Comment (caguero) [inline mbzirc_ign/src/GameLogicPlugin.cc:1012]: Update comment?

**Review by caguero — COMMENTED** [review]
It works for me as expected besides very minor comments.

File: mbzirc_ign/src/GameLogicPlugin.cc (hunk 3 above)
Comment (iche033) [reply mbzirc_ign/src/GameLogicPlugin.cc]: oops, updated. [2140ea8](https://github.com/osrf/mbzirc/pull/60/commits/2140ea86bd7af26ef48f14c75ad7df155b0f2e19)

File: mbzirc_ign/src/GameLogicPlugin.cc (hunk 2 above)
Comment (iche033) [reply mbzirc_ign/src/GameLogicPlugin.cc]: fixed. [2140ea8](https://github.com/osrf/mbzirc/pull/60/commits/2140ea86bd7af26ef48f14c75ad7df155b0f2e19)

File: mbzirc_ign/src/GameLogicPlugin.cc (hunk 1 above)
Comment (iche033) [reply mbzirc_ign/src/GameLogicPlugin.cc:271]: added. [2140ea8](https://github.com/osrf/mbzirc/pull/60/commits/2140ea86bd7af26ef48f14c75ad7df155b0f2e19)

**Review by caguero — APPROVED** [review]

**Pull request merged**